src/CentrifugeRouter.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.21;
+
+import {Auth} from "./Auth.sol";
+import {SafeTransferLib} from "./libraries/SafeTransferLib.sol";
+import "./interfaces/IERC7540.sol";
+
+interface IERC7540ExtLike {
+    function deposit(uint256 assets, address receiver, address owner) external returns (uint256 shares);
+}
+
+interface IERC20Like {
+    function approve(address spender, uint256 value) external returns (bool);
+}
+
+contract CentrifugeRouter is Auth {
+    mapping(address => mapping(address => uint256)) public lockedRequests;
+
+    // --- Interactions ---
+
+    // --- Deposit ---
+    function requestDeposit(address vault, uint256 amount) external {
+        SafeTransferLib.safeTransferFrom(IERC7540(vault).asset(), msg.sender, address(this), amount);
+        IERC20Like(IERC7540(vault).asset()).approve(vault, amount); // option to add approve function, make poolManager
+            // ward and create global approval on vault deployment
+        IERC7540(vault).requestDeposit(amount, msg.sender, address(this), "");
```

Comment:
This should work, because the router is endorsed (aka automatically an operator):
```suggestion
        IERC7540(vault).requestDeposit(amount, msg.sender, msg.sender, "");
```